### PR TITLE
Fix JAVA table crashes (#6992)

### PR DESCRIPTION
**Pull request by LeonarddeR**
Technical background for this has been provided in #6992. I tested this wit the application provided by @legle, as well as with the SQuirreL SQL Client.

Note that this does not fix the issue that JABHandler.getHWNDFromAccessibleContext sometimes results in a crash. of the java application. According to @jcsteh, this is probably a bug in the JAVA access bridge.

### CODE REVIEW

**Comment by jcsteh** [comment]
@leonardder, I haven't reviewed this yet, as I'd like to get an answer on https://github.com/nvaccess/nvda/issues/6992#issuecomment-300036077 first. Thanks.

**Comment by leonardder** [comment]
@JCSTeh: I answered your question in https://github.com/nvaccess/nvda/issues/6992#issuecomment-301690132

Please let me know whether there are there any open ends left before you can start reviewing. This issue has a quite major impact on the usage of JAVA applications, so it would be great if we could manage it to have this in 2017.3

File: source/JABHandler.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
+# -*- coding: UTF-8 -*-
 #javaAccessBridgeHandler.py
 #A part of NonVisual Desktop Access (NVDA)
-#Copyright (C) 2006-2007 NVDA Contributors <http://www.nvda-project.org/>
+#Copyright (C) 2007-2017 NV Access Limited, Peter Vágner, Renaud Paquay, Babbage B.V.
```
Comment (jcsteh) [inline source/JABHandler.py]: I updated this myself, as it was missing some previous contributors and it's a bit tricky to get the Unicode stuff right for Peter's name. Hope you don't mind.

**Review by jcsteh — CHANGES_REQUESTED** [review]
Your explanation in https://github.com/nvaccess/nvda/issues/6992#issuecomment-301690132 seems reasonable. This looks fine to me. However, please add a comment above each of these blocks (where you start passing hwnd) explaining why you pass hwnd. Without this, it might be difficult to figure out why we think tables are special in future. Something like this:

```
# #6992: Querying the hwnd for table related objects can cause the app to crash.
# A table is almost certainly contained within a single hwnd,
# so just pass the hwnd for the querying object.
```

I realise this means a bit of duplication, but I'd rather duplication than accidentally regressing this in future because we didn't realise it was significant.

File: source/JABHandler.py (hunk 1 above)
Comment (leonardder) [inline source/JABHandler.py]: No problem at all, thanks for this!

**Review by jcsteh — APPROVED** [review]

**Pull request merged**